networkx/algorithms/community/tests/test_louvain.py

```diff
@@ -43,6 +43,58 @@ def test_partition():
     assert part == partition
 
 
+def test_directed_partition():
+    """
+    Test 2 cases that were looping infinitely
+    from issues #5175 and #5704
+    """
+    G = nx.DiGraph()
+    H = nx.DiGraph()
```

Comment:
The original failure reported in #5175 was also sensitive to node insertion order, at least by my testing (see [this comment](https://github.com/networkx/networkx/issues/5175#issuecomment-965602312)) so I think we also need:

```suggestion
    H = nx.DiGraph()
    G.add_nodes_from(range(10))
```

Reply:
I'm going to go ahead and put this one in, just to verify that the tests still pass afterwards!